### PR TITLE
Add data docs

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigration.cs [hunk 1]
```diff
@@ -2,8 +2,14 @@
 
 namespace OrchardCore.Data.Migration
 {
+    /// <summary>
+    /// Represents a contract for a database migrations.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigration.cs]: migrations => migration

File: src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs [hunk 2]
```diff
@@ -3,6 +3,9 @@
 
 namespace OrchardCore.Data.Migration
 {
+    /// <summary>
+    /// Represents a contract for manage database migrations.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs]: to manage

File: src/OrchardCore/OrchardCore.Data.Abstractions/ISessionHelper.cs [hunk 3]
```diff
@@ -3,18 +3,26 @@
...
+    /// Represents a contract that provides a helper methods for <see cref="YesSql.ISession"/>.
+    /// </summary>
     public interface ISessionHelper
     {
         /// <summary>
         /// Loads a single document (or create a new one) for updating and that should not be cached.
         /// For a full isolation, it needs to be used in pair with <see cref="GetForCachingAsync"/>.
         /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="factory">A factory methods to loads or creates a document.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Data.Abstractions/ISessionHelper.cs]: to load and create

File: src/OrchardCore/OrchardCore.Data.Abstractions/ISessionHelper.cs [hunk 4]
```diff
@@ -3,18 +3,26 @@
...
         Task<T> LoadForUpdateAsync<T>(Func<T> factory = null) where T : class, new();
 
         /// <summary>
         /// Gets a single document (or create a new one) for caching and that should not be updated.
         /// For a full isolation, it needs to be used in pair with <see cref="LoadForUpdateAsync"/>.
         /// </summary>
+        /// <typeparam name="T">The.</typeparam>
+        /// <param name="factory">A factory method to gets or creates a document for caching.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Data.Abstractions/ISessionHelper.cs]: to get or create

File: src/OrchardCore/OrchardCore.Data/Migration/DataMigrationManager.cs [hunk 5]
```diff
@@ -13,6 +13,9 @@
 
 namespace OrchardCore.Data.Migration
 {
+    /// <summary>
+    /// Represents a class that manage the database migrations.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Data/Migration/DataMigrationManager.cs]: that manages

File: src/OrchardCore/OrchardCore.Data/SessionHelper.cs [hunk 6]
```diff
@@ -5,21 +5,25 @@
 
 namespace OrchardCore.Data
 {
+    /// <summary>
+    /// Represents a class that provides a helper methods for the <see cref="ISession"/>.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Data/SessionHelper.cs]: a helper method => helper method

**Comment by hishamco** [comment]
Can we merge this?

**Comment by hishamco** [comment]
@agriffard could you please merge this?

**Pull request merged**